### PR TITLE
Make the AccessibleDialog dismissible via X icon + consolidate styling

### PR DESCRIPTION
**Pull request by Erin007**
I realized that we currently have [11 use cases](https://github.com/search?q=repo%3Acode-dot-org%2Fcode-dot-org+AccessibleDialog&type=code) of the `AccessibleDialog` component. Most of those render an X close icon and then [duplicate the same css](https://github.com/search?q=repo%3Acode-dot-org%2Fcode-dot-org%20.xCloseButton%20&type=code) in individual scss modules. Let's bake the X icon into the component!

## Testing story

I manually compared the local versions of each use case with production 

| Component | After | Difference | 
| ------------- | ------------- | ------------- |
| `ChatWarningModal` used in GenAI unit + AI Tutor| chat_warning_modal | none |
| `CompareModelsDialog` used in GenAI unit|compare_models | none |
| `DanceAiModal` used to select emojis| dance_ai_modal | none - this one had so much custom styling already that I just hid the baked in X icon |
| `ExtraLinksModal`  used in lab2 for levelbuilders| extra_links_modal | none |
| `MoreDetailsDialog` used on the teacher dashboard to show progress key|progress_more_details| none |
| `InviteToV2ProgressModal` used on the teacher dashboard| invite_to_v2| none |
| `RemoveCoteacherDialog` used in section management| remove co-teacher| now dismissible via X icon |
| `MultipleSectionsAssigner` used on the course catalogue |multiple section assigner| now dismissible via X icon |
| `NoSectionsToAssignDialogs` used on the course catalogue |no section assigner| now dismissible via X icon |
| `HourOfCodeGuideEmailDialog` used on ... | hoc_email| now dismissible via X icon |

Python Lab `GraphModal` can be seen here: https://github.com/code-dot-org/code-dot-org/pull/59517

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgressV2/InviteToV2ProgressModal.jsx [hunk 1]
```diff
@@ -134,15 +134,6 @@ function InviteToV2ProgressModal({
...
         >
-          <button
-            id="ui-close-dialog"
-            type="button"
-            onClick={handleDismiss}
-            aria-label={i18n.closeDialog()}
-            className={styles.xCloseButton}
-          >
-            <i id="x-close" className="fa-solid fa-xmark" />
-          </button>
```
Comment (Erin007) [inline apps/src/templates/sectionProgressV2/InviteToV2ProgressModal.jsx:145]: This modal makes a distinction between dismissing the dialog and closing the dialog and then logs those separate events for analytics. 
- [x] check with Teacher Tools about this and see if implementing a workaround is needed

**Comment by Erin007** [comment]
- [x] remove the report abuse dialog + tendrils as follow up 
See https://github.com/code-dot-org/code-dot-org/pull/59614

**Comment by Erin007** [comment]
https://github.com/code-dot-org/code-dot-org/pull/59662 Was merged while this work as in flight. Once #59560 is merged, an X icon will appear on that dialog too. 
- [ ] post-merge double check everything is as expected

**Comment by Erin007** [comment]
fyi @sanchitmalhotra126 @cearachew I touched the shared ai chat Warning Modal with this change. There shouldn't be any user impact but let me know if something looks off. Thanks!

**Review by sanchitmalhotra126 — COMMENTED** [review]
DSCO also has a `CloseButton` component now https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/componentLibrary/closeButton/CloseButton.tsx - not sure if the styling matches up exactly, but just FYI in case you're able to use it!

File: apps/src/templates/sectionProgressV2/InviteToV2ProgressModal.jsx [hunk 2]
```diff
@@ -69,30 +69,34 @@ function InviteToV2ProgressModal({
...
   }, [dateProgressTableInvitationDelayed, hasSeenProgressTableInvite]);
 
-  const handleModalClose = React.useCallback(() => {
-    analyticsReporter.sendEvent(EVENTS.PROGRESS_V2_SEEN_INVITATION, {
-      sectionId,
-    });
+  const closeModal = () => {
     setInvitationOpen(false);
-  }, [sectionId]);
+  };
```
Comment (Erin007) [inline apps/src/templates/sectionProgressV2/InviteToV2ProgressModal.jsx:74]: @kobryan0619 does this still get you the logging you need?

Comment (kobryan0619) [reply apps/src/templates/sectionProgressV2/InviteToV2ProgressModal.jsx:74]: It does.

**Review by kobryan0619 — COMMENTED** [review]
The InviteToV2Modal looks good to me!

**Review by sanchitmalhotra126 — APPROVED** [review]
🧹 🎉 great cleanup!

**Pull request merged**